src/ImageSharp/Memory/ByteMemoryOwner{T}.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers;
+
+namespace SixLabors.ImageSharp.Memory
+{
+    /// <summary>
+    /// A custom <see cref="IMemoryOwner{T}"/> that can wrap <see cref="IMemoryOwner{T}"/> of <see cref="byte"/> instances
+    /// and cast them to be <see cref="IMemoryOwner{T}"/> for any arbitrary unmanaged <typeparamref name="T"/> value type.
+    /// </summary>
+    /// <typeparam name="T">The value type to use when casting the wrapped <see cref="IMemoryOwner{T}"/> instance.</typeparam>
+    internal sealed class ByteMemoryOwner<T> : IMemoryOwner<T>
+        where T : unmanaged
+    {
+        private readonly IMemoryOwner<byte> memoryOwner;
+        private readonly ByteMemoryManager<T> memoryManager;
+        private bool disposedValue;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ByteMemoryOwner{T}"/> class.
+        /// </summary>
+        /// <param name="memoryOwner">The <see cref="IMemoryOwner{T}"/> of <see cref="byte"/> instance to wrap.</param>
+        public ByteMemoryOwner(IMemoryOwner<byte> memoryOwner)
+        {
+            this.memoryOwner = memoryOwner;
+            this.memoryManager = new ByteMemoryManager<T>(memoryOwner.Memory);
+        }
+
+        /// <inheritdoc/>
+        public Memory<T> Memory => this.memoryManager.Memory;
+
+        private void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
```

Comment:
This is not thread safe, is this an issue?
We might want to replace this with `Interlocked.CompareExchange` and make the flag an `int` to enable this, but I'm not sure if the rest of the library actually gives strong thread-safety guarantees on other APIs anyway, so not entirely sure whether this is needed. I mean, wouldn't hurt either though, I guess 🤔
@JimBobSquarePants thoughts?